### PR TITLE
drivers: ieee802154: Use nRF Secure Services to read EUI64

### PR DESCRIPTION
**Pull request by czeslawmakarski**
This commit allows the setting of ieee802154 EUI64 address
in the non-secure build by reading the FICR registers out
of the secure zone.

Signed-off-by: Czeslaw Makarski <Czeslaw.Makarski@nordicsemi.no>

### CODE REVIEW

File: drivers/ieee802154/ieee802154_nrf5.c [hunk 1]
```diff
@@ -88,7 +97,13 @@ static void nrf5_get_eui64(uint8_t *mac)
 
 #if defined(CONFIG_SOC_NRF5340_CPUAPP) && \
 	defined(CONFIG_TRUSTED_EXECUTION_NONSECURE)
-#error Accessing EUI64 on the non-secure mode is not supported at the moment
+	int ret = spm_request_read(&factoryAddress, (uint32_t)EUI64_ADDR,
+							sizeof(factoryAddress));
```
Comment (tbursztyka) [inline drivers/ieee802154/ieee802154_nrf5.c:101]: re-indent, it should be below last '(' + 1 space.

**Comment by ioannisg** [comment]
I think this PR should be sent downstream @czeslawmakarski, not in the Zephyr Project, unless I am missing something.

**Review by ioannisg — CHANGES_REQUESTED** [review]
I think this is a candidate for downstream. Please advice otherwise.

**Comment by czeslawmakarski** [comment]
@ioannisg You are right. I am retargeting now.

**Pull request closed without merging**